### ISSUE TITLE
Compact Machines Lagging Server

### ISSUE THREAD

**Issue by DrWAM** [CompactMods/CompactMachines#521]
### I am not using performance mods.

- [X] Confirm

### Description

I treat this mod like my own personal TARDIS. My character will enter and leave at will as long as theyre able to get back. Each time I enter or exit the dimension my server lags out and no player can interact with the world, dimensions, blocks, mobs.

Occasionally when reentering the dimension I will load in on the outside of the box and fall endlessly until I'm kicked from the server.

I've tried hosting this server on multiple devices with varying Linux Distros. Each server has been given up to 16G of ram and eight cores which is way more than any server should need.

### Mod Version

4.3.0

### Forge Version

1.18.2 - 4.1.31

### Screenshot (if available)

![image_209](https://user-images.githubusercontent.com/39708259/171674594-f8a76a1e-6c3f-4eaf-a45d-0f1b995f95e4.png)

### How to reproduce

1.) Install mod
2.) Give player compact machine and remote
3.) Profit?

**Comment by robotgryphon** [comment]
Please provide a modpack or mod list so I can properly profile the mod; this will be hard to reproduce in the small environment I usually test in. Thank you!

**Comment by robotgryphon** [comment]
Adding stale tag; if no response is received by 8/3 this will be closed

**Comment by Twisted-Code** [comment]
@robotgryphon Don't know if you want to reopen this or just want me to create a new issue, but I've been having the same  issue as OP (That awkward feeling when you ping... a spotify dev?... who happens to have the username "OP". Sorry Orjan Fors!) on All The Mods 8 (ATM8). One of my friends rented a server from Akliz (no idea the details of their subscription plan, though I will say it doesn't seem to be fully virtualized machine as I can see the RAM usage of other subscribers), which I've been helping him optimize and troubleshoot, and we noticed every time someone enters or leaves the dimension, everyone lags and sometimes even gets timed out. My current working hypothesis is that the dimension is completely unloading and reloading because we only have a few players and nothing else happening in the dimension, but I'm not really sure where to go from there.
I can probably run the spark profiler for you, but first, do you want me to make a new issue?

**Comment by robotgryphon** [comment]
@macks2008 I can reopen this issue, sure. Please do attach profiler data if you have it!

**Comment by Twisted-Code** [comment]
Okay, will do possibly tomorrow, I don't want to go disrupting the server's performance right now while other people are using it.

**Comment by Twisted-Code** [comment]
https://spark.lucko.me/fUJM7kRmiM

**Comment by Twisted-Code** [comment]
One thing I should add: If I were reporting this, I would not be able to check the "I am not using performance mods" task, at least not without trying again without Rubidium. In a pack of this size, that is impractical at best since it takes forever to load already... Sorry about that!
Also, when I tested to do the above profile, the lag spike wasn't nearly as bad as the first time. It was still pretty noticeable when I went into the machine, but it was fine when inside the machine and leaving it. I think the main performance improvement to be made, and our main complaint, is that, when going into the machine, the server completely froze while it loaded the dimension. This doesn't happen when we teleport to the Nether or End when no one is there, so what are you doing different? Could you possibly pre-load the dimension whenever someone is holding a Personal Shrinking Device?

**Comment by Twisted-Code** [comment]
(Basically you shouldn't be stalling the server tick while terrain loads. I think that's what's going on for me)

**Comment by Twisted-Code** [comment]
Not to rush you, but what's the status on fixing this so it can load the dimension in parallel with the server tick (which appears to be the root of the problem for me)?

**Comment by Twisted-Code** [comment]
...ok then. I hope that's just because the issue is stale, but I suppose I can open a new issue either way if it occurs in 1.21+?
